Make iced::Result Send + Sync

With the recent addition of error handling, I tried to wrap an iced::Result with an anyhow::Result, but it's not working because I get this error:

```
error[E0277]: `(dyn StdError + 'static)` cannot be sent between threads safely                                                                                                                                                                                               
  --> src\main.rs:51:23
   |
51 |     App::run(settings)?;
   |                       ^ `(dyn StdError + 'static)` cannot be sent between threads safely
   |
   = help: the trait `std::marker::Send` is not implemented for `(dyn StdError + 'static)`
   = note: required because of the requirements on the impl of `std::marker::Send` for `Unique<(dyn StdError + 'static)>`
   = note: required because it appears within the type `Box<(dyn StdError + 'static)>`
   = note: required because it appears within the type `iced::Error`
   = note: required because of the requirements on the impl of `From<iced::Error>` for `anyhow::Error`
   = note: required by `std::convert::From::from`

error[E0277]: `(dyn StdError + 'static)` cannot be shared between threads safely
  --> src\main.rs:51:23
   |
51 |     App::run(settings)?;
   |                       ^ `(dyn StdError + 'static)` cannot be shared between threads safely
   |
   = help: the trait `Sync` is not implemented for `(dyn StdError + 'static)`
   = note: required because of the requirements on the impl of `Sync` for `Unique<(dyn StdError + 'static)>`
   = note: required because it appears within the type `Box<(dyn StdError + 'static)>`
   = note: required because it appears within the type `iced::Error`
   = note: required because of the requirements on the impl of `From<iced::Error>` for `anyhow::Error`
   = note: required by `std::convert::From::from`
```

Example code:

```rs
use anyhow::Result;

fn main() -> Result<()> {
    use iced::{Application, Settings};

    App::run(Settings::default())?;
    Ok(())
}
```

--- a/src/error.rs
+++ b/src/error.rs
@@ -9,7 +9,7 @@ pub enum Error {
 
     /// The application window could not be created.
     #[error("the application window could not be created")]
-    WindowCreationFailed(Box<dyn std::error::Error>),
+    WindowCreationFailed(Box<dyn std::error::Error + Send + Sync>),
 
     /// A suitable graphics adapter or device could not be found.
     #[error("a suitable graphics adapter or device could not be found")]
@@ -30,5 +30,16 @@ impl From<iced_winit::Error> for Error {
                 Error::GraphicsAdapterNotFound
             }
         }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn assert_send_sync() {
+        fn _assert<T: Send + Sync>() {}
+        _assert::<Error>();
     }
 }

